scripts/tools/zap/zap_download.py

```diff
@@ -141,19 +141,19 @@ def _GetZapVersionToUse(project_root):
     # This heuristic may be bad at times, however then you can also override the
     # version in command line parameters
 
-    match_re = re.compile(r'.*ENV\s+ZAP_VERSION=([^# ]*)')
-
-    docker_path = os.path.join(project_root, "integrations/docker/images/chip-build/Dockerfile")
-
-    with open(docker_path, 'rt') as f:
-        for l in f.readlines():
-            l = l.strip()
-            m = match_re.match(l)
-            if not m:
-                continue
-            return m.group(1)
-
-    raise Exception(f"Failed to determine version from {docker_path}")
+    zap_version = ""
+    zap_path = os.path.join(project_root, "scripts/setup/zap.json")
+    zap_json = json.load(open(zap_path))
+    for package in zap_json.get("packages", []):
+        for tag in package.get("tags", []):
+            if tag.startswith("version:2@"):
+                zap_version = tag[10:]
```

Comment:
Might be clearer and more error-prone if you use `removeprefix` here, as long as we can assume Python 3.9 or newer.